pkgs/tools/package-management/dnf5/default.nix

```diff
@@ -0,0 +1,95 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, createrepo_c
+, gettext
+, help2man
+, pkg-config
+, cppunit
+, fmt
+, glib
+, json_c
+, libmodulemd
+, libpeas
+, librepo
+, libsmartcols
+, libsolv
+, libxml2
+, rpm
+, sdbus-cpp
+, sqlite
+, systemd
+, toml11
+, zchunk
+}:
+
+
+stdenv.mkDerivation rec {
+  pname = "dnf5";
+  version = "5.0.15";
+
+  src = fetchFromGitHub {
+    owner = "rpm-software-management";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-0MR9CJDFL1vbuO7FZyyn3PNb0p27oaho6I2eminTyYU=";
+  };
+
+  nativeBuildInputs = [ cmake createrepo_c gettext help2man pkg-config ];
+  buildInputs = [
+    cppunit
+    fmt
+    glib
+    json_c
+    libmodulemd
+    libpeas
+    librepo
+    libsmartcols
+    libsolv
+    libxml2
+    rpm
+    sdbus-cpp
+    sqlite
+    systemd
+    toml11
+    zchunk
+  ];
+
+  meta = with lib; {
+    description = "Next-generation RPM package management system";
+    homepage = "https://github.com/rpm-software-management/dnf5";
+    license = licenses.gpl2Plus;
+    maintainers = with lib.maintainers; [ malt3 ];
+    platforms = platforms.linux ++ platforms.darwin;
+  };
```

Comment:
Move this block at the very end of the file, just like pretty much all the others derivations in `nixpkgs`.